src/realm/sync/noinst/client_reset_recovery.cpp

```diff
@@ -934,18 +961,18 @@ void RecoverLocalChangesetsHandler::operator()(const Instruction::Update& instr)
         }
         Status on_list_index(LstBase& list, uint32_t index) override
         {
-            util::Optional<ListTracker::CrossListIndex> cross_index;
-            cross_index = m_recovery_applier->m_lists.at(m_list_path).update(index);
-            if (cross_index) {
-                m_instr.prior_size = static_cast<uint32_t>(list.size());
-                m_instr.path.back() = cross_index->remote;
-            }
-            else {
+            auto cross_index = m_recovery_applier->m_lists.at(m_list_path).update(index);
+            if (!cross_index) {
                 return Status::DidNotResolve;
             }
+            m_instr.prior_size = static_cast<uint32_t>(list.size());
```

Comment:
Do we have a test that hits this?

Reply:
we don't apparently. i'll update one of the many existing ones since it's pretty easy to hit it.